src/frontend/screens/LeaveProject/index.tsx

```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import { useState } from "react";
+import { defineMessages, FormattedMessage } from "react-intl";
+import {
+  NavigationStackProp,
+  NavigationStackScreenComponent,
+  NavigationStackScreenProps,
+} from "react-navigation-stack";
+import { LeaveProjectCompleted } from "./LeaveProjectCompleted";
+import { LeaveProjectInitial } from "./LeaveProjectInitial";
+import { LeaveProjectProgress } from "./LeaveProjectProgess";
+import HeaderTitle from "../../sharedComponents/HeaderTitle";
+
+const m = defineMessages({
+  headerTitle: {
+    id: "screens.LeaveProject.LeaveProject",
+    defaultMessage: "Leave Project",
+  },
+});
+
+enum ScreenStates {
+  initial,
+  progress,
+  completed,
+}
+
+export interface ILeaveSharedProp {
+  screenStateHook: [
+    ScreenStates,
+    React.Dispatch<React.SetStateAction<ScreenStates>>
+  ];
+}
+
+const LeaveProjectScreen: NavigationStackScreenComponent = ({
+  navigation,
+  screenProps,
+  theme,
+}: NavigationStackScreenProps) => {
+  const { initial, progress, completed } = ScreenStates;
+  const [screen, setScreen] = useState(initial);
+
+  if (screen === initial) {
+    return <LeaveProjectInitial screenStateHook={[screen, setScreen]} />;
+  }
+
+  if (screen === progress) {
+    return <LeaveProjectProgress screenStateHook={[screen, setScreen]} />;
+  }
+
+  return <LeaveProjectCompleted screenStateHook={[screen, setScreen]} />;
```

Comment:
this works but seems a little odd to me. i would just provide a callback as the prop that's specific to the screen i.e.

 ```tsx
const goToNextScreen = () => setScreen(previousScreen => previousScreen + 1)
...
<LeaveProjectInitial next={goToNextScreen} />
<LeaveProjectProgress next={goToNextScreen} />
...
```

Reply:
Yeah, I am unecessarily passing props here, I will change it!